lib/instana/util.rb

```diff
@@ -161,6 +161,24 @@ def header_to_id(given)
         return '' unless given.match(/\A[a-z\d]{16,32}\z/i)
         given
       end
+
+      # Get the redis url from redis client
+      #
+      # @param client [String] the Redis client
+      #
+      # @return [String, nil]
+      #
+      def get_redis_url(client)
+        if client.respond_to?(:config) # redis-client
```

Comment:
Are the things in the comment different redis libraribes?

I would probably also add which check to which redis version is related.

Reply:
These are different redis client libraries. Until 2017 Sidekiq used `hi-redis`, after that they switched to `redis-rb` and recently in version 7 they started using `redis-client`